great_expectations/expectations/metrics/column_aggregate_metrics/column_value_counts.py

```diff
@@ -2,6 +2,27 @@
 
 import pandas as pd
 
+from great_expectations.compatibility.pyspark import (
+    DataFrame as pyspark_sql_DataFrame,
```

Comment:
One of the main goals of breaking these compatibility files up into package specific files (like pyspark.py) is so we don't need to add the prefixes like `pyspark_sql_` in this case (unless there is a conflict).

Reply:
@anthonyburdi This was a matter of style -- in particular, I wanted to underscore it for modules that can have multiple `ExecutionEngine` implementations (e.g., most metrics).  So this was a move for "let us be super strict and careful" (even though we could "get away" with being a bit more lax -- I feel that this was would reduce confusion long term).  Thanks.

Reply:
How about instead, now that we have the `pyspark` module we do `from great_expectations.compatibility import pyspark` and then in usage `pyspark.DataFrame`. This would make it clear which DataFrame we are using in metric implementations.
This style is also compatible with our style guide: https://google.github.io/styleguide/pyguide.html#22-imports which is to "Use import statements for packages and modules only, not for individual classes or functions." (The link to the style guide gives more info for motivation and other cases).
It would be nice to move towards that style for all of our code over time, and especially here since we are changing so many import statements.

Reply:
@anthonyburdi The idea is nice, but I am not seeing how it will work, because `DataFrame` is imported not from `pyspark` but from `pyspark.sql`.  This situation is not limited to `DataFrame` in `pyspark` -- there are many other occurrences across the packages we are importing.  Do you have a solution in mind?  Thanks.

Reply:
In usages we import the new pyspark compatibility package (`compatibility/pyspark.py` via `from great_expectations.compatibility import pyspark`) which imports `DataFrame`. Then in metrics etc we use that `DataFrame` from the `compatibility/pyspark.py`. Here is an example:

`great_expectations/compatibility/pyspark.py`:
```
try:
    from pyspark.sql import DataFrame
except (ImportError, AttributeError):
    DataFrame = SPARK_NOT_IMPORTED
```

then in usage:

```
from great_expectations.compatibility import pyspark

my_df: pyspark.DataFrame
```

Reply:
@anthonyburdi My approach focused on not having users/developers import `SPARK_NOT_IMPORTED` and other similar "`_NOT_IMPORTED`" constants.  I was pursuing the idea of "things just working" in application modules, whereby all "`_NOT_IMPORTED`" constants would be treated as "internal implementation" and stayed encapsulated within the appropriate compatibility modules.  I was concerned that types and other issues could happen if users had to follow the try/except pattern and import/utilize the "`_NOT_IMPORTED`" constants in application modules.  Happy to discuss.  Thank you.

Reply:
@anthonyburdi I affected the change as per our conversation.  Thanks!